Lib/multiprocessing/reduction.py

```diff
@@ -248,34 +273,61 @@ def _rebuild_socket(df, family, type, proto):
     register(socket.socket, _reduce_socket)
 
 
-class AbstractReducer(metaclass=ABCMeta):
+class AbstractReducer(metaclass=abc.ABCMeta):
     '''Abstract base class for use in implementing a Reduction class
     suitable for use in replacing the standard reduction mechanism
     used in multiprocessing.'''
-    ForkingPickler = ForkingPickler
-    register = register
-    dump = dump
     send_handle = send_handle
     recv_handle = recv_handle
 
     if sys.platform == 'win32':
-        steal_handle = steal_handle
-        duplicate = duplicate
+        steal_handle = staticmethod(steal_handle)
+        duplicate = staticmethod(duplicate)
         DupHandle = DupHandle
     else:
-        sendfds = sendfds
-        recvfds = recvfds
-        DupFd = DupFd
+        sendfds = staticmethod(sendfds)
+        recvfds = staticmethod(recvfds)
+        DupFd = staticmethod(DupFd)
 
-    _reduce_method = _reduce_method
-    _reduce_method_descriptor = _reduce_method_descriptor
-    _rebuild_partial = _rebuild_partial
-    _reduce_socket = _reduce_socket
-    _rebuild_socket = _rebuild_socket
+    _reduce_method = staticmethod(_reduce_method)
+    _reduce_method_descriptor = staticmethod(_reduce_method_descriptor)
+    _rebuild_partial = staticmethod(_rebuild_partial)
+    _reduce_socket = staticmethod(_reduce_socket)
+    _rebuild_socket = staticmethod(_rebuild_socket)
 
     def __init__(self, *args):
         register(type(_C().f), _reduce_method)
         register(type(list.append), _reduce_method_descriptor)
         register(type(int.__add__), _reduce_method_descriptor)
         register(functools.partial, _reduce_partial)
         register(socket.socket, _reduce_socket)
+
+    @property
+    def HAVE_SEND_HANDLE(self):
+        import sys
+        return (sys.platform == 'win32' or
+                (hasattr(socket, 'CMSG_LEN') and
+                 hasattr(socket, 'SCM_RIGHTS') and
+                 hasattr(socket.socket, 'sendmsg')))
+
+    def get_pickler_class(self):
+        return AbstractPickler
+
+    def get_unpickler_class(self):
+        return AbstractUnpickler
+
+    def register(self, type_, reduce_):
```

Comment:
This method (a) doesn't appear to be used and (b) doesn't necessarily work, because nothing requires the `get_pickler_class` to define a `register` method.

Reply:
Regarding (b), this is stated in the documentation that `get_pickler_class` should return a subclass of an `AbstractPickler`, that itself (as a subclass of `ForkingPickler`), implements a `register` method. Regarding (a), same comment as above (this @pablogsal's work), but I'll think about it myself if this function can be removed.